runtime/common/src/migrations.rs

```diff
@@ -178,43 +180,42 @@ where
 	}
 }
 
+const DEMOCRACY_PALLET: &'static str = "Democracy";
 parameter_types! {
-	pub const CouncilPalletName: &'static str = "Council";
-	pub const TechnicalCommitteePalletName: &'static str = "TechnicalCommittee";
+	pub const DemocracyPalletName: &'static str = DEMOCRACY_PALLET;
 }
 
-pub struct PalletCollectiveDropGovV1Collectives<Runtime>(pub PhantomData<Runtime>);
-impl<Runtime> Migration for PalletCollectiveDropGovV1Collectives<Runtime>
+pub struct RemovePalletDemocracy<Runtime>(pub PhantomData<Runtime>);
+impl<Runtime> Migration for RemovePalletDemocracy<Runtime>
 where
 	Runtime: frame_system::Config,
 {
 	fn friendly_name(&self) -> &str {
-		"MM_RemoveGovV1Collectives"
+		"MM_RemoveDemocracyPallet"
 	}
 
 	fn migrate(&self, _available_weight: Weight) -> Weight {
-		log::info!("Removing Council and Tech from pallet_collective");
-
-		let mut weight = Weight::zero();
-
-		let w = frame_support::migrations::RemovePallet::<
-			CouncilPalletName,
-			<Runtime as frame_system::Config>::DbWeight,
-		>::on_runtime_upgrade();
-		weight = weight.saturating_add(w);
-
-		let w = frame_support::migrations::RemovePallet::<
-			TechnicalCommitteePalletName,
+		log::info!("Removing pallet democracy");
+
+		let hashed_prefix = twox_128(DEMOCRACY_PALLET.as_bytes());
+		if contains_prefixed_key(&hashed_prefix) {
```

Comment:
I do not understand the point of this check, `frame_support::migrations::RemovePallet` below only does anything if there are keys in the storage.

If the storage is small enough, we can just remove this condition and let the migration remove the whole storage. If there are a considerable amount of keys, we can add a lazy migration.

Reply:
we do have already the lazy migration here https://github.com/moonbeam-foundation/moonbeam/pull/2651 this check is to make sure that in case the lazy migration hasn't been completed for some reason we do not risk to stall the chain

Reply:
@noandrea the lazy migration remove only the `VotingOf` storage item, so your check should verify the prefix `twox128("Democracy) ++ twox128("VotingOf")`instead